### ISSUE TITLE
feature request. standalone encrypted database

### ISSUE THREAD

**Issue by nutpantz** [SimpleMobileTools/Simple-Contacts#70]
The standard android database is queried by so many different apps that your contacts list is hardly secure. It would be nice if simple contacts had its own separate encrypted database for private contacts you don't want every other app to see.

**Comment by tibbi** [comment]
a private database has been added, hidden from other apps. I dont plan encrypting it, but I will leave the ticket open anyway.

**Comment by nutpantz** [comment]
I will test it as soon as I can get the update from fdroid

**Comment by tibbi** [comment]
it was actually added in 3.2.0 already released 10 days ago, I just didnt write it here

**Comment by nutpantz** [comment]
How do I import vcf to the phone storage contacts? I exported from my contact list then tried to import, it calls it successful but nothing is added to phone storage

**Comment by nutpantz** [comment]
It appears that when trying to import it created vnd.sec.contact.phone and added the contacts there..

**Comment by tibbi** [comment]
at selecting the target contact source you should see an item called "Phone storage (not visible by other apps)", select that one.

**Comment by nutpantz** [comment]
Yep that's what I selected.. It still created vnd.sec.contact.phone I'll try to delete that some how and try again

**Comment by nutpantz** [comment]
Ok got it, when importing if there is nothing else on device it just shows phone storage and it appears like that's where its going. But you have to manually select it anyways or it creates vnd.contacts.

**Comment by tibbi** [comment]
ye I can reproduce your case only on an emulator, maybe Ill improve it someday. For now it isnt worth spending too much time on. Once you select a contact source at creating a contact it will be remembered.

**Comment by nutpantz** [comment]
It's still a huge improvement in privacy over using the build in database.. I'm a happy clam.. And I'll only harass you about encrypting it once pour twice a year.

**Comment by sefernet** [comment]
Encryption would be awesome!  
Great work ! 
Regards..

**Comment by olistik** [comment]
Even if it's not encrypted it could be useful to decide its location.

A really good use case for people that don't want to rely on a Google account to sync their contact list between devices (mobile/desktop) could be to set the location of the database (or even the location of a `vcf` replica that gets updated each time the database receives an update) into a path that is used by sharing app like [Syncthing](https://syncthing.net/) that performs end-to-end encryption, making the encryption a lesser urgent matter.

**Comment by olistik** [comment]
I see that there is an [sqlite `contacts.db`](https://github.com/SimpleMobileTools/Simple-Contacts/blob/27e1366339b4fb89e6074c8af7c9daba7a31f442/app/src/main/kotlin/com/simplemobiletools/contacts/helpers/DBHelper.kt#L55) stored somewhere but I can't find it when I look for it with a file manager app under `/Android/data/com.simplemobiletools.contacts/`.

It is available only when I turn `adb root` and then I'm able to pull it from `/data/data/com.simplemobiletools.contacts/databases/contacts.db`.

Would you find it useful to provide a way to either make it visible or move a copy of it elsewhere?

**Comment by olistik** [comment]
@tibbi Do you think the above mentioned feature request deserves an issue on its own?

**Comment by tibbi** [comment]
ye create a different issue for that please. Currently the contacts.db is stored at the internal app space, Im not really modifying its path.

**Comment by olistik** [comment]
For reference, I created it here https://github.com/SimpleMobileTools/Simple-Contacts/issues/275

**Comment by BEEDELLROKEJULIANLOCKHART** [comment]
@sefernet and @nutpantz, why not encrypt your storage-device rather than rely ypon the encryptive method of this software? Surely the former is more secure, considering that any encryption that Simple Contacts performs is probably reversible by it.
